### PR TITLE
Add Coil CNI to networking.md

### PR DESCRIPTION
**Pull request by ymmt2005**
This is to add an external link to a new CNI plugin [Coil](https://github.com/cybozu-go/coil) to `networking.md`.

Coil is an open-source CNI plugin that features IPv4/v6, multiple address pools, egress NAT gateways, and more.
Here is an introductory blog article about Coil: https://blog.kintone.io/entry/coilv2

### CODE REVIEW

File: content/en/docs/concepts/cluster-administration/networking.md [hunk 1]
```diff
@@ -158,6 +158,11 @@ tables to provide per-instance subnets to each host (which is limited to 50-100
 entries per VPC). In short, cni-ipvlan-vpc-k8s significantly reduces the
 network complexity required to deploy Kubernetes at scale within AWS.
 
+### Coil
+
+[Coil](https://github.com/cybozu-go/coil) is a CNI plugin designed for easy-integration and flexible egress networking.
+Coil provides bare metal performance and allows you to define arbitrary egress NAT gateways for external networks.
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/networking.md:164]:
```suggestion
Coil operates with a low overhead compared to bare metal, and allows you to define arbitrary egress
NAT gateways for external networks.
```
I'm inferring that Coil has _some_ overhead but that it's really low - does that seem fair?

File: content/en/docs/concepts/cluster-administration/networking.md [hunk 2]
```diff
@@ -158,6 +158,11 @@ tables to provide per-instance subnets to each host (which is limited to 50-100
 entries per VPC). In short, cni-ipvlan-vpc-k8s significantly reduces the
 network complexity required to deploy Kubernetes at scale within AWS.
 
+### Coil
+
+[Coil](https://github.com/cybozu-go/coil) is a CNI plugin designed for easy-integration and flexible egress networking.
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/networking.md:163]: (nit)
```suggestion
[Coil](https://github.com/cybozu-go/coil) is a CNI plugin designed for ease of integration, providing flexible
egress networking.
```

**Review by sftim — COMMENTED** [review]
/lgtm

It's clear why this link to 3rd party documentation is appropriate.
I did have a couple of suggestions.

File: content/en/docs/concepts/cluster-administration/networking.md (hunk 1 above)
Comment (ymmt2005) [reply content/en/docs/concepts/cluster-administration/networking.md:164]: Yes, thank you.  I will update.

File: content/en/docs/concepts/cluster-administration/networking.md (hunk 2 above)
Comment (ymmt2005) [reply content/en/docs/concepts/cluster-administration/networking.md:163]: Thanks.

**Comment by ymmt2005** [comment]
@sftim 
Thanks for your review.  I have updated the PR.

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-25194--kubernetes-io-master-staging.netlify.app/docs/concepts/cluster-administration/networking/#how-to-implement-the-kubernetes-networking-model

**Pull request merged**